thunder/executors/torch_autograd.py

```diff
@@ -236,7 +236,18 @@ def split_forward_backward(computation_trc: TraceCtx, compile_data, compile_stat
                 compile_data.fn.bucketing_strategy != FSDPBucketingStrategy.NONE,
             )
         if getattr(compile_data.fn, "sharding_strategy") == FSDPType.ZERO2:
-            fw_extrace = sort_waits(fw_extrace)
+            from thunder.distributed import FSDPBucketingStrategy
+            from thunder.distributed.utils import limit_in_flight_allgathers
+            from sys import maxsize as INT_MAX
+
+            # sort the allgather+wait as consumer order just before consumer
+            fw_extrace = sort_waits_for_zero3(fw_extrace)
```

Comment:
Naming is difficult but it would be a good service for others reading the code to rename this function since it's now used in the "if ZERO2" block.